monkey/infection_monkey/master/ip_scanner.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import queue
+import threading
+from queue import Queue
+from threading import Event
+from typing import Callable, Dict, List
+
+from infection_monkey.i_puppet import IPuppet, PingScanData, PortScanData
+
+from .threading_utils import create_daemon_thread
+
+logger = logging.getLogger()
+
+Callback = Callable[[str, PingScanData, Dict[int, PortScanData]], None]
+
+
+class IPScanner:
+    def __init__(self, puppet: IPuppet, num_workers: int):
+        self._puppet = puppet
+        self._num_workers = num_workers
+
+    def scan(self, ips_to_scan: List[str], options: Dict, results_callback: Callback, stop: Event):
+        # Pre-fill a Queue with all IPs to scan so that threads know they can safely exit when the
+        # queue is empty.
+        ips = Queue()
+        for ip in ips_to_scan:
+            ips.put(ip)
+
+        scan_ips_args = (ips, options, results_callback, stop)
+        scan_threads = []
+        for i in range(0, self._num_workers):
+            t = create_daemon_thread(target=self._scan_ips, args=scan_ips_args)
```

Comment:
Type-hinting for `create_daemon_thread` throws me a warning: 
```
Expected type 'Tuple[Any]', got 'Tuple[Queue, dict, (str, PingScanData, Dict[int, PortScanData]) -> None, Event]' instead
```
`Any` stands for any type, not "any number of any types". The type hint should either just be `Any` or we need to  use structural typing